lib/joint/WheelJoint.ts

```diff
@@ -0,0 +1,690 @@
+  }
+
+  /**
+   * Get the anchor point on bodyB in world coordinates.
+   * 
+   * @return {Vec2}
+   */
+  getAnchorB() {
+    return this.m_bodyB.getWorldPoint(this.m_localAnchorB);
+  }
+
+  /**
+   * Get the reaction force on bodyB at the joint anchor in Newtons.
+   * 
+   * @param {float} inv_dt
+   * @return {Vec2}
+   */
+  getReactionForce(inv_dt) {
+    return Vec2.combine(this.m_impulse, this.m_ay, this.m_springImpulse, this.m_ax).mul(inv_dt);
+  }
+
+  /**
+   * Get the reaction torque on bodyB in N*m.
+   * 
+   * @param {float} inv_dt
+   * @return {float}
+   */
+  getReactionTorque(inv_dt) {
+    return inv_dt * this.m_motorImpulse;
+  }
+
+  initVelocityConstraints(step) {
+    this.m_localCenterA = this.m_bodyA.m_sweep.localCenter;
+    this.m_localCenterB = this.m_bodyB.m_sweep.localCenter;
+    this.m_invMassA = this.m_bodyA.m_invMass;
+    this.m_invMassB = this.m_bodyB.m_invMass;
+    this.m_invIA = this.m_bodyA.m_invI;
+    this.m_invIB = this.m_bodyB.m_invI;
+
+    var mA = this.m_invMassA;
+    var mB = this.m_invMassB; // float
+    var iA = this.m_invIA;
+    var iB = this.m_invIB; // float
+
+    var cA = this.m_bodyA.c_position.c;
+    var aA = this.m_bodyA.c_position.a;
+    var vA = this.m_bodyA.c_velocity.v;
+    var wA = this.m_bodyA.c_velocity.w;
+
+    var cB = this.m_bodyB.c_position.c;
+    var aB = this.m_bodyB.c_position.a;
+    var vB = this.m_bodyB.c_velocity.v;
+    var wB = this.m_bodyB.c_velocity.w;
+
+    var qA = Rot.neo(aA);
+    var qB = Rot.neo(aB);
+
+    // Compute the effective masses.
+    var rA = Rot.mulVec2(qA, Vec2.sub(this.m_localAnchorA, this.m_localCenterA));
+    var rB = Rot.mulVec2(qB, Vec2.sub(this.m_localAnchorB, this.m_localCenterB));
+    var d = Vec2.zero();
+    d.addCombine(1, cB, 1, rB);
+    d.subCombine(1, cA, 1, rA); // Vec2
+
+    // Point to line constraint
+    {
+      this.m_ay = Rot.mulVec2(qA, this.m_localYAxisA);
+      this.m_sAy = Vec2.cross(Vec2.add(d, rA), this.m_ay);
+      this.m_sBy = Vec2.cross(rB, this.m_ay);
+
+      this.m_mass = mA + mB + iA * this.m_sAy * this.m_sAy + iB * this.m_sBy
+          * this.m_sBy;
+
+      if (this.m_mass > 0.0) {
+        this.m_mass = 1.0 / this.m_mass;
+      }
+    }
+
+    // Spring constraint
+    this.m_springMass = 0.0;
+    this.m_bias = 0.0;
+    this.m_gamma = 0.0;
+    if (this.m_frequencyHz > 0.0) {
+      this.m_ax = Rot.mulVec2(qA, this.m_localXAxisA);
+      this.m_sAx = Vec2.cross(Vec2.add(d, rA), this.m_ax);
+      this.m_sBx = Vec2.cross(rB, this.m_ax);
+
+      var invMass = mA + mB + iA * this.m_sAx * this.m_sAx + iB * this.m_sBx
+          * this.m_sBx; // float
+
+      if (invMass > 0.0) {
+        this.m_springMass = 1.0 / invMass;
+
+        var C = Vec2.dot(d, this.m_ax); // float
+
+        // Frequency
+        var omega = 2.0 * Math.PI * this.m_frequencyHz; // float
+
+        // Damping coefficient
+        var d = 2.0 * this.m_springMass * this.m_dampingRatio * omega; // float
```

Comment:
We can't use `let d` here because the outer `d` is also used in line 499 and I did not have the courage to come up with another name 😄